localstack/services/ec2/provider.py

```diff
@@ -0,0 +1,206 @@
+    ) -> DescribeReservedInstancesResult:
+        return DescribeReservedInstancesResult(
+            ReservedInstances=[
+                ReservedInstances(
+                    AvailabilityZone="eu-central-1a",
+                    Duration=2628000,
+                    End=datetime(2016, 6, 30, tzinfo=timezone.utc),
+                    FixedPrice=0.0,
+                    InstanceCount=2,
+                    InstanceType=InstanceType.t2_small,
+                    ProductDescription=RIProductDescription.Linux_UNIX,
+                    ReservedInstancesId=long_uid(),
+                    Start=datetime(2016, 1, 1, tzinfo=timezone.utc),
+                    State=ReservedInstanceState.active,
+                    UsagePrice=0.05,
+                    CurrencyCode=CurrencyCodeValues.USD,
+                    InstanceTenancy=Tenancy.default,
+                    OfferingClass=OfferingClassType.standard,
+                    OfferingType=OfferingTypeValues.Partial_Upfront,
+                    RecurringCharges=[
+                        RecurringCharge(Amount=0.05, Frequency=RecurringChargeFrequency.Hourly)
+                    ],
+                    Scope=scope.Availability_Zone,
+                )
+            ]
+        )
+
+    @handler("PurchaseReservedInstancesOffering", expand=False)
+    def purchase_reserved_instances_offering(
+        self,
+        context: RequestContext,
+        purchase_reserved_instances_offerings_request: PurchaseReservedInstancesOfferingRequest,
+    ) -> PurchaseReservedInstancesOfferingResult:
+        return PurchaseReservedInstancesOfferingResult(
+            ReservedInstancesId=long_uid(),
+        )
+
+    @handler("DeleteVpcEndpoints", expand=False)
+    def delete_vpc_endpoints(
+        self, context: RequestContext, delete_vpc_endpoints_request: DeleteVpcEndpointsRequest
+    ) -> DeleteVpcEndpointsResult:
+        return moto.call_moto(context)
+
+    @handler("CreateVpcEndpointServiceConfiguration")
+    def create_vpc_endpoint_service_configuration(
+        self,
+        context: RequestContext,
+        dry_run: Boolean = None,
+        acceptance_required: Boolean = None,
+        private_dns_name: String = None,
+        network_load_balancer_arns: ValueStringList = None,
+        gateway_load_balancer_arns: ValueStringList = None,
+        client_token: String = None,
+        tag_specifications: TagSpecificationList = None,
+    ) -> CreateVpcEndpointServiceConfigurationResult:
+        region = Ec2Backend.get()
+
+        service_id = short_uid()
+        service_config = ServiceConfiguration(
+            ServiceId=service_id,
+            AcceptanceRequired=acceptance_required,
+            PrivateDnsName=private_dns_name,
+            NetworkLoadBalancerArns=network_load_balancer_arns,
+            GatewayLoadBalancerArns=gateway_load_balancer_arns,
+        )
+        region.vpc_endpoint_service_configurations[service_id] = service_config
```

Comment:
The `vpc_endpoint_service_configurations` field is not (yet) defined in our `Ec2Backend` region backend. In the previous version of the provider, we were attaching it to the moto `ec2_backend` like this: 

```
        configs = getattr(self.ec2_backend, "vpc_endpoint_service_configurations", {})
        self.ec2_backend.vpc_endpoint_service_configurations = configs
``` 
This looks a bit clunky, but I think it is preferable, mostly as it makes our lives easier for persistence support in -ext. Can we attach this field, and also `prefix_lists` to the moto `ec2_backend` directly?

Reply:
It turns out this is implemented in Moto

https://github.com/spulec/moto/blob/0fca96c/moto/ec2/_models/vpc_service_configuration.py#L53-L84

Removed in [eae3a83](https://github.com/localstack/localstack/pull/5608/commits/eae3a831d33e6c317e66889d94d06ca3defaca83)